### ISSUE TITLE
Incorrect syllable splitting in the dictionary pinyin search

### ISSUE THREAD

**Issue by pdg137** [Pin1yin1/pin1yin1#16]
Entering "xiong" into the dictionary pinyin search results in a search for "xi o ng". Other syllables starting with "xi" split properly, so maybe it's just unaware of that one.

**Comment by fifieldt** [comment]
confirmed

**Comment by fifieldt** [comment]
Also applies to qiong

**Comment by fifieldt** [comment]
Also jiong.

It looks like the common theme is that these are written 'xiong' in pinyin, but 'xiung' in bopomofo

**Comment by fifieldt** [comment]
bug is here:

```
 if(bopomofo[current_syllable+"1"] != undefined &&
         bopomofo[new_syllable+"1"] == undefined &&
         current_syllable != "")
      {
        // adding the character would stop a valid syllable
        // so add a space!
        new_val += " "
        current_syllable = new_letter
      }
```

since "xio" isn't a valid syllable, and "xi" is, the javascript inserts a space when it shouldn't.

**Comment by fifieldt** [comment]
Slight bug with the new code - 

if numbered tones are used, an extra space is inserted at the end of the string

eg

chuang3dang4jiang1hu2 - won't match 闖 蕩 江 湖
chuang3 - will only find words that have at least two characters, ignoring eg 刱

**Comment by pdg137** [comment]
Thanks - fixed that in commit c710e3f8.
